### ISSUE TITLE
race when returning call.shared

### ISSUE THREAD

**Issue by vroldanbet** [janos/singleflight#7]
Go race detector identified the following race

```
==================
WARNING: DATA RACE
Write at 0x00c00092b2f0 by goroutine 55:
  resenje.org/singleflight.(*Group[go.shape.string,go.shape.*uint8]).Do()
      /go/pkg/mod/resenje.org/singleflight@v0.4.0/singleflight.go:43 +0x440
  REDACTED

Previous read at 0x00c00092b2f0 by goroutine 43:
  resenje.org/singleflight.(*Group[go.shape.string,go.shape.*uint8]).wait()
      /go/pkg/mod/resenje.org/singleflight@v0.4.0/singleflight.go:86 +0x1d0
  resenje.org/singleflight.(*Group[go.shape.string,go.shape.*uint8]).Do()
      /go/pkg/mod/resenje.org/singleflight@v0.4.0/singleflight.go:47 +0x4b0
  REDACTED
```

I believe this is happening because [`call.shared` is returned in `Group.wait` without synchronization](https://github.com/janos/singleflight/blob/a3a78ab136065cbea2e1f147c8ab6438ad56a585/singleflight.go#L86) while it can be [written concurrently in `Group.Do`](https://github.com/janos/singleflight/blob/a3a78ab136065cbea2e1f147c8ab6438ad56a585/singleflight.go#L43).

https://github.com/janos/singleflight/blob/a3a78ab136065cbea2e1f147c8ab6438ad56a585/singleflight.go#L86

https://github.com/janos/singleflight/blob/a3a78ab136065cbea2e1f147c8ab6438ad56a585/singleflight.go#L43

**Comment by janos** [comment]
Thank you for reporting the issue and for providing the details.

I was able to reproduce it and data race is fixed in version [0.4.1](https://github.com/janos/singleflight/releases/tag/v0.4.1).

**Comment by vroldanbet** [comment]
Thanks for the quick turnaround, @janos! 🙏🏻
